py/desiutil/depend.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+
+"""
+Utilities for working with code dependencies stored in FITS headers.
+
+The code name and the code version are stored in pairs of keywords, similar
+to how table columns are defined.  e.g.
+
+DEPNAM00 = 'numpy'
+DEPVER00 = '1.11'
+DEPNAM01 = 'desiutil'
+DEPVER01 = '1.4.1'
+
+The functions and Dependencies class provided here provide convenience
+wrappers to loop over they keywords looking for a particular dependency
+and adding a new dependency version to next available DEPNAMnn/DEPVERnn.
+
+Examples:
+
+>>> import desiutil
+>>> from desiutil import depend
+>>> import astropy
+>>> from astropy.io import fits
+>>> 
+>>> hdr = fits.Header()
+>>> depend.setdep(hdr, 'desiutil', desiutil.__version__)
+>>> depend.setdep(hdr, 'astropy', astropy.__version__)
+>>> depend.getdep(hdr, 'desiutil')
+'1.4.1.dev316'
+>>> depend.hasdep(hdr, 'astropy')
+True
+>>> hdr
+DEPNAM00= 'desiutil'
+DEPVER00= '1.4.1.dev316'
+DEPNAM01= 'astropy '
+DEPVER01= '1.1.1   '
+
+There is also an object wrapper that gives a header dict-like semantics to
+update or view dependencies.  This directly updates the input header object
+so that it can be used in subsequent I/O
+
+>>> codever = depend.Dependencies(hdr)
+>>> codever['blat'] = '1.2'
+>>> codever['foo'] = '3.4'
+>>> for codename, version in codever.items():
+...     print(codename, version)
+... 
+('desiutil', '1.4.1.dev316')
+('astropy', u'1.1.1')
+('blat', '1.2')
+('foo', '3.4')
+
+"""
+
+def setdep(header, name, version):
+    '''
+    set dependency version for code name
+
+    Args:
+        header : dict-like object to store dependencies
+        name : code name string
+        version : version string
+    '''
+    for i in range(100):
+        namekey = 'DEPNAM{:02d}'.format(i)
+        verkey = 'DEPVER{:02d}'.format(i)
+        if namekey in header:
+            if header[namekey] == name:
+                header[namekey] = version
```

Comment:
I'm fixing this myself on the next commit, but I just wanted to point out that this wouldn't have worked.